File: 338.counting-bits.kt
// @leet start
class Solution {
    fun countBits(n: Int): IntArray {
      val ans = IntArray(n+1)
      var offset = 1
      for (i in 1..n) {
        if (i == offset * 2) offset *= 2
        ans[i] = ans[i - offset] + 1
      }
      return ans
    }
}
// @leet end

File: 338.counting-bits.go
// @leet start
func countBits(n int) []int {
  ans := make([]int, n+1)
  offset := 1
  for i := 1; i <= n; i++ {
    if (i == offset*2) { offset *= 2 }
    ans[i] = 1 + ans[i-offset]
  }
  return ans
}
// @leet end
